enhancements/insights/container-scanner.md

```diff
@@ -0,0 +1,499 @@
+
+### Container Scanner Test Plan
+
+The container scanner can be tested in isolation to verify its capability to scan containers and collect the expected runtime info.
+
+Testing the container scanner can be done outside of the Insights Operator test suite.
+It requires integration tests with applications based on the detectable runtime stacks.
+
+### Insights Operator Test Plan
+
+The integration tests for the Insights Operator will verify that it is able to properly deploy the container scanner as a `DaemonSet` and have some smoke tests to validate that the additional runtime info can effectively be collected.
+
+## Graduation Criteria
+
+TBD
+
+### Dev Preview -> Tech Preview
+
+This enhancement would be behind the featuregate capabilities of OpenShift
+
+* Add a non-public flag to enable it on a cluster to collect runtime information
+* End User documentation is not required
+* Sufficient test coverage
+* Gather feedback from users
+* Generate reports to validate the consumption of the additional runtime information
+
+### Tech Preview -> GA
+
+*  More testing
+*  Sufficient time for feedback
+*  Available by default
+*  Conduct load testing
+* User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/blob/main/support/remote_health_monitoring/using-insights-operator.adoc) if needed
+
+### Removing a deprecated feature
+
+Not relevant for this proposal
+
+## Upgrade / Downgrade Strategy
+
+Upgrades/downgrades of the container scanner will be handled by updating its image URL in the Insights Operator configuration.
+The release cycle of the container scanner is bound to the Insight Operator release cycle.
+
+This proposal does not bring additional change to the upgrade/downgrade of the OpenShift Insights Operator.
+
+## Version Skew Strategy
+
+This proposal does not impact the version skew strategy of the Insights Operator.
+
+## Operational Aspects of API Extensions
+
+The container scanner runs with its own cgroups and does not impact the scanned containers.
+
+However, it will consume a certain amount of CPU and memory on each worker node while it is scanning.
+These consumptions must be bounded to avoid disrupting the user workload.
+
+Scanning containers will also increase the time to collect insights by the operator. The workload gatherer is performed every 12 hours and its execution will depend on the number of containers that are scanned.
+	
+In the existing initial implementation of the container scanner, it takes __~180 ms to scan a single container__.
+
+---
+**TBD** Provide measures of resource consumption (memory and CPU)
+
+---
+
+## Support Procedures
+
+The failure of scanning containers must not impact the ability of the Insights operator to finish collecting workload data.
+
+Failures to deploy the container-scanner will be reported as an error in the `"workloads/workload_info"` stanza of the `insights-operator/gathers.json` file inside the archive generated by the operator.
```

Comment:
I think topics like who will maintain the container-scanner, who will setup CI, etc. should be part of the discussion.